src/components/AuthForm/ForgotPasswordPage.jsx

```diff
@@ -0,0 +1,235 @@
+			const { status } = await verifyEmail(email);
+			console.log("isEmailFound :", status);
+
+			if (status === 200) {
+				setForgotPasswordState(prev => ({
+					...prev,
+					isEmailVerified: true
+				}));
+				showToast("success", "An OTP sent successfully to your Email..");
+			} else {
+				showToast("error", "Could'nt find your email..");
+			}
+		} catch (error) {
+			console.error(error);
+		} finally {
+			setForgotPasswordState(prev => ({ ...prev, isVerifying: false }));
+		}
+	};
+
+	const onVerifyOTP = async event => {
+		event.preventDefault();
+
+		if (!otp || otp.length < 6) {
+			showToast("error", "Please Enter Valid Otp!");
+			return;
+		}
+
+		try {
+			setForgotPasswordState(prev => ({ ...prev, isVerifying: true }));
+
+			const response = await verifyOTP(email, otp);
+
+			if (response.status === 200) {
+				showToast("success", "OTP verified successfully!");
+				setForgotPasswordState(prev => ({
+					...prev,
+					isOTPVerified: true,
+					verificationToken: response.data.token
+				}));
+			} else if (response.code === 401) {
+				showToast("error", "Invalid OTP..");
+			}
+		} catch (error) {
+			throw new Error(error);
+		} finally {
+			setForgotPasswordState(prev => ({ ...prev, isVerifying: false }));
+		}
+	};
+
+	const onResendOtp = async () => {
+		try {
+			const { status } = await verifyEmail(email);
+
+			if (status === 200) {
+				setForgotPasswordState(prev => ({
+					...prev,
+					isEmailVerified: true
+				}));
+			} else {
+				showToast("error", "Could'nt find your email..");
+			}
+		} catch (error) {
+			throw new Error(error);
+		}
+	};
+
+	const onNewPasswordSubmit = async event => {
+		event.preventDefault();
+
+		if (passwordError || confirmPasswordError) {
+			showToast("error", "Invalid Password");
+			return;
+		} else if (password !== confirmPassword) {
+			showToast("error", "Password and confirm Password must match..");
+			return;
+		}
+
+		try {
+			setLoaderText("Updating password");
+			setForgotPasswordState(prev => ({
+				...prev,
+				isVerifying: true
+			}));
+			const { status } = await setNewPassword(email, password, verificationToken);
+			if (status === 200) {
+				showToast("success", "Password Changed Successfully..");
+				navigate("/auth");
+			}
+		} catch (error) {
+			throw new Error(error);
+		} finally {
+			setForgotPasswordState(prev => ({ ...prev, isVerifying: false }));
+		}
+	};
+
+	const loader = (
+		<div className="loader-wrapper">
+			<h1 className="text">{loaderText}</h1>{" "}
+			<ReactLoading
+				type="balls"
+				color="#fff"
+				className="balls-loader"
+			/>
+		</div>
+	);
+
+	return (
+		<div id="forgot-password-page-container">
+			<VerifyEmail
+				emailValue={email}
+				onChange={handleInputChange}
+				emailError={emailError}
+				onEmailSubmit={onVerifyEmail}
+			/>
+			{isEmailVerified && (
+				<VerifyOTP
+					otpValue={otp}
+					onChangeOtp={handleInputChange}
+					onVerify={onVerifyOTP}
+					resendOtp={onResendOtp}
+				/>
```

Comment:
@Ayush-Bharadva Whatever I have said in my previous comment for `<VerifyEmail>` component applies here as well. only `handleNext` prop. 

All the OTP related states, API calls and concerns will stay inside `<VerifyOTP>` and will not be exposed to it via props.